### PR TITLE
Rename intellectual property publisher

### PR DESCRIPTION
**Pull request by nancy-mejia**
# Description

Rename `intellectual_property_publisher` to `intellectual_property` in the YAML files we use for registry operations.
Registry entity property will remain the same.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/registry/registry.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ class RegistrySchema(ResourceSchema):
         casing_transform=snake_to_pascal,
     )
     replication_locations = fields.List(NestedField(RegistryRegionDetailsSchema))
-    intellectual_property_publisher = fields.Str()
+    intellectual_property = fields.Str()
```
Comment (shivamMg) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/registry/registry.py:33]: Do we need to change this? - from my understanding it should either be IntellectualProperty  object or we should keep it as is (intellectual_property_publisher string). cc @vikasagrawal-ms

Comment (nancy-mejia) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/registry/registry.py]: I have updated this to be an object to match the way we should define the YAMLs from the spec.

**Review by hugoaponte — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/registry/registry.py [hunk 2]
```diff
@@ -30,7 +31,7 @@ class RegistrySchema(ResourceSchema):
         casing_transform=snake_to_pascal,
     )
     replication_locations = fields.List(NestedField(RegistryRegionDetailsSchema))
-    intellectual_property_publisher = fields.Str()
+    intellectual_property = NestedField(IntellectualPropertySchema)
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/registry/registry.py:34]:
```suggestion
    intellectual_property = ExperimentalField(NestedField(IntellectualPropertySchema))
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_registry/registry.py [hunk 3]
```diff
@@ -189,6 +189,11 @@ def _convert_yaml_dict_to_entity_input(
             if global_acr_exists:
                 if not hasattr(region_detail, "acr_details") or len(region_detail.acr_details) == 0:
                     region_detail.acr_config = [acr_input]
+        if INTELLECTUAL_PROPERTY in input:
+            intellectual_property = input.pop(INTELLECTUAL_PROPERTY)
+            publisher = intellectual_property.get("publisher", None)
+            if publisher:
+                input["intellectual_property_publisher"] = publisher
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_registry/registry.py:196]: have we verified that we want this difference between SDK/CLI contracts? Once we GA both, we cannot change them.

**Review by needuv — CHANGES_REQUESTED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_registry/registry.py (hunk 3 above)
Comment (needuv) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_registry/registry.py:196]: synced offline. Given registry GA has been delayed, we can address this later.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/registry/registry.py (hunk 2 above)
Comment (needuv) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/registry/registry.py:34]: Not needed if adding back experimental decorator

**Review by needuv — APPROVED** [review]

**Pull request merged**